doc: add unittest core dump instructions (#22226)
bitcoin/bitcoin: src/test/README.md

```bash

> gdb src/test/test_bitcoin

Reviewer: On a second thought. Won't this be enough to observe segmentation faults in gdb if they appear? Is there a reason to take the extra step of a core dump?

Author: Yeah, this is a good point; willfix.

```

#### Segmentation faults

> By default, the boost test runner will intercept system errors and not produce a core

Reviewer 2: ```suggestion
By default, the Boost test runner will intercept system errors and not produce a core
```

file. To bypass this, add `--catch_system_errors=no` to the `test_bitcoin` arguments
and ensure that your ulimits are set properly (e.g. `ulimit -c unlimited`).

Running the tests and hitting a segmentation fault should now produce a file called `core`

> depending upon your `/proc/sys/kernel/core_pattern` contents (on Linux platforms).

Reviewer 2: ```suggestion
(on Linux platforms, the file name will depend on the contents of `/proc/sys/kernel/core_pattern`).
```

```bash
gdb src/test/test_bitcoin
```

#### Segmentation faults

By default, the boost test runner will intercept system errors and not produce a core
file. To bypass this, add `--catch_system_errors=no` to the `test_bitcoin` arguments
and ensure that your ulimits are set properly (e.g. `ulimit -c unlimited`). 

Running the tests and hitting a segmentation fault should now produce a file called `core`

> (depending upon your `/proc/sys/kernel/core_pattern` contents).

Reviewer 3: Non-blocking nit: Could clarify that ``/proc/sys/kernel/core_pattern` is Linux specific :)

Author: Fixed, thanks

```bash
gdb src/test/test_bitcoin
```

#### Segmentation faults

If you hit a segmentation fault during a test run, you can diagnose where the fault
is happening by running `gdb ./src/test/test_bitcoin` and then using the `bt` command
within gdb.

Another tool that can be used to resolve segmentation faults is
[valgrind](https://valgrind.org/).

If for whatever reason you want to produce a core dump file for this fault, you can do

> that as well. By default, the boost test runner will intercept system errors and not

Reviewer 2: ```suggestion
that as well. By default, the Boost test runner will intercept system errors and not
```

produce a core file. To bypass this, add `--catch_system_errors=no` to the
`test_bitcoin` arguments and ensure that your ulimits are set properly (e.g. `ulimit -c
unlimited`).

Running the tests and hitting a segmentation fault should now produce a file called `core`

> (on Linux platforms, the file name will likely depend on the contents of 

Reviewer 2: linter appeasement
```suggestion
(on Linux platforms, the file name will likely depend on the contents of
```

`/proc/sys/kernel/core_pattern`).

You can then explore the core dump using
``` bash

> gdb src/test/test_bitcoin core

Reviewer 2: ```suggestion
gdb ./src/test/test_bitcoin core
```

Reviewer 2: (you write `gdb ./src/test/test_bitcoin` above. note that this is the same instruction as at the top of the diff)
